Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,361 @@
+
+    /**
+     * Creates the QueueMessage object.
+     *
+     * @return QueueMessageInterface
+     */
+    public function createQueueMessage(): QueueMessageInterface
+    {
+        $msg = new QueueMessage();
+        $msg->setContext(new Context());
+
+        return $msg;
+    }
+
+    /**
+     * Verifies if there is some connection opened with the broker.
+     */
+    public function isConnected(): bool
+    {
+        foreach ($this->amqpConnections as $connection) {
+            if ($connection->isConnected()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function destroy()
+    {
+        $this->disconnect();
+    }
+
+    /**
+     * Consumes the message and dispatch it to others features.
+     *
+     * @param string $consumerTag
+     * @param string $queueName
+     * @param callable|null $callback
+     */
+    public function consume(string $consumerTag, string $queueName, callable $callback = null)
+    {
+        $this->channel->basic_consume($queueName, $consumerTag, false, false, false, false, $callback);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function ack(int $messageId = null)
+    {
+        $this->channel->basic_ack($messageId);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function nack(int $messageId = null)
+    {
+        $this->channel->basic_nack($messageId);
+    }
+
+    /**
+     * Creates a message and prepare the content to publish.
+     *
+     * {@inheritdoc}
+     *
+     * @throws \Exception
+     */
+    public function send(QueueMessageInterface $message, $destination = null, array $arguments = [])
+    {
+        $this->declareChannel();
+        $this->validateQueueExists($message);
+        $messageBody = $this->getSerializer()->serialize($message, $this->format);
+        $messageHeaders = new AMQPTable($message->getHeaders());
+        $amqpMessage = new AMQPMessage($messageBody);
+        $amqpMessage->set('application_headers', $messageHeaders);
+        $this->declareQueue($message->getQueue(), AMQP_DURABLE, $arguments);
+        $this->channel->basic_publish($amqpMessage, self::EXCHANGE_NAME, $message->getQueue());
+        return true;
+    }
+
+    /**
+     * Declares the queue to drive the message.
+     *
+     * @param string $name      The name of the queue
+     * @param array  $arguments See AMQPQueue::setArguments()
+     *
+     * @return array|null
+     *
+     * @throws \AMQPChannelException
+     */
+    public function declareQueue(string $queueName, int $durable = AMQP_DURABLE, array $arguments = [])
+    {
+        return $this->channel->queue_declare($queueName, false, $durable, false, false, false, new AMQPTable([$arguments]));
+    }
+
+    /**
+     * Catch a channel inside the connection.
+     */
+    public function declareChannel(): AMQPChannel
+    {
+        try {
+            $this->channel = $this->amqpConnections[0]->channel();
```

Comment:
In this case the channel is always created using the amqpConnections, key 0 but amqpConnection is an array where you are specifying different connections

Reply:
Yes. I did an implementation of different channel with a same connection. But, the channel was getting lost every time on the flow. We need to do a little refactory on the future if we need to use the approach of multiple channels. The reason for this is because we might to keep the channnel_id related for each process, otherwise you not must to be able to do anything related to the driver. Because of the complexity and time do implement this, Andres and I decided to postpone this feature.

Reply:
If this is a possible future feature, I wouldn't have half implementation